cylc/flow/ws_data_mgr.py

```diff
@@ -364,9 +422,12 @@ def update_task_proxies(self, task_ids=None):
                 prereq_obj = prereq.api_dump(self.workflow_id)
                 if prereq_obj:
                     prereq_list.append(prereq_obj)
+            del tproxy.prerequisites[:]
             tproxy.prerequisites.extend(prereq_list)
+            outputs = []
             for _, msg, is_completed in itask.state.outputs.get_all():
-                tproxy.outputs.append(f"{msg}={is_completed}")
+                outputs.append(f"{msg}={is_completed}")
+            tproxy.outputs[:] = outputs
```

Comment:
In Python `append` and `for` are slow, list comprehensions are fast:

```python
import time

N = 10000000

# for loop
x = []
a = time.time()
for i in range(N):
    x.append(N)
print(time.time() - a)

# list comp
a = time.time()
x = [i for i in range(N)]
print(time.time() - a)
```

```bash
$ python test.py
1.185086727142334
0.37612366676330566
```

So this would probably take about 1/3 the time (and less memory as there is no intermediate list) as the for loop equivalent:

```python
outputs = [
    f"{msg}={is_completed}"
    for _, msg, is_completed in itask.state.outputs.get_all()
]
```

Reply:
This is so annoying, because the `for` loop is so much more readable.

Reply:
> This is so annoying, because the for loop is so much more readable.

I have come to disagree with this, when spread over multiple lines with each line starting with "if" or "for" I actually find this clearer.

Reply:
It is certainly a very English way of expressing things, as I was told when I first learnt *Perl*.

Reply:
Mattism #1 - everything boils down to Perl!

Reply:
Fixed.